packages/grid/_modules_/grid/hooks/features/filter/useFilter.ts

```diff
@@ -106,25 +110,19 @@ export const useFilter = (apiRef: ApiRef, rowsProp: RowsProp): void => {
     [apiRef, forceUpdate, logger, setGridState],
   );
 
-  const applyFilters = React.useCallback(
-    (noRerender = false) => {
-      if (options.filterMode === FeatureModeConstant.server) {
-        return;
-      }
+  const applyFilters = React.useCallback(() => {
+    if (options.filterMode === FeatureModeConstant.server) {
+      return;
+    }
 
-      clearFilteredRows();
+    clearFilteredRows();
 
-      const { items, linkOperator } = apiRef.current.state.filter;
-      items.forEach((filterItem) => {
-        applyFilter(filterItem, linkOperator);
-      });
-
-      if (!noRerender) {
-        forceUpdate();
-      }
-    },
-    [apiRef, applyFilter, clearFilteredRows, forceUpdate, options.filterMode],
-  );
+    const { items, linkOperator } = apiRef.current.state.filter;
+    items.forEach((filterItem) => {
```

Comment:
Shouldn't we use the `setGridState` to update the state?

Reply:
Just applying filters here not updating the state

Reply:
If it is not updating the state then why do you need the `forceUpdate()` after it?

Reply:
it's updating the state in the applyFilter below, using apiRef avoid an extra rendering to update the rows with the latest filter.